src/script/interpreter.cpp

```diff
@@ -396,16 +395,18 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
 
     try
     {
-        while (pc < pend)
+        while (pc.size() > 0)
         {
             bool fExec = vfExec.all_true();
 
             //
             // Read instruction
             //
-            if (!script.GetOp(pc, opcode, vchPushValue))
+            Span<const unsigned char> push_value;
```

Comment:
Do we get the benefit of the `Span` here even though it's only used in a short context?

Reply:
Not sure what you mean by "the benefit of the `Span`". Spans are a more readable way of dealing with span-like data, but it depends what you care about, and what you compare with.

Compared to the current code, it's more efficient: it's avoiding an allocation and a copy (vchPushValue was storing a copy of the pushed data). A copy still happens when emplacing on the stack, but only in executed branches.

Compared to manually maintaining a pointer & length, which would be equally efficient... it's certainly a lot more readable.

Reply:
I see, thanks